src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -5130,6 +5130,33 @@ int RGWRados::delete_bucket(RGWBucketInfo& bucket_info, RGWObjVersionTracker& ob
     (void) CLSRGWIssueBucketIndexClean(index_pool.ioctx(),
 				       bucket_objs,
 				       cct->_conf->rgw_bucket_index_max_aio)();
+  } else {
+    bucket_info.flags |= BUCKET_DELETED;
+    static constexpr auto max_retries = 10;
+    int retries = 0;
+    do {
+      r = ctl.bucket->store_bucket_instance_info(bucket, bucket_info, y, dpp, RGWBucketCtl::BucketInstance::PutParams()
+                                                                      .set_exclusive(false)
+                                                                      .set_mtime(real_time())
+                                                                      .set_attrs(&attrs)
+                                                                      .set_orig_info(&bucket_info));
+      if (r == -ECANCELED) {
+        //racing write. re-read bucket info
+        map<string, bufferlist> attrs;
+        int ret = get_bucket_instance_info(bucket, bucket_info, nullptr, &attrs, y, dpp);
```

Comment:
shouldn't we reuse the existing `attrs` instead of reading and throwing away this local `attrs` variable?

Reply:
yes, no reason to throw it away. done!